### PR TITLE
Adds validation for the examples

### PR DESCRIPTION
**Pull request by elijahbenizzy**
All examples must adhere to a strict format. We'll be adding a directory with "unstructured" examples that don't adhere.

TODO on this:
- [x] Fix up all examples
    - [x] conversational-rag
    - [x] hello-world-counter
    - [x] llm-adventure-game
    - [x] multi-agent-collaboration
    - [x] multi-modal-chatbot
    - [x] simple-chatbot-intro
    - [x] streaming-overview

### CODE REVIEW

File: examples/validate_examples.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+# Use pytest_generate_tests to dynamically parameterize the fixture
+def pytest_generate_tests(metafunc):
+    # if "directory" in metafunc.fixturenames:
+    directories = get_directories(EXAMPLES_DIR)
+    metafunc.parametrize("directory", directories, scope="module")
+
+
+def test_directory_contents(directory):
+    required_files = ["notebook.ipynb", "application.py", "README.md", "digraph.png"]
```
Comment (skrawcz) [inline examples/validate_examples.py:41]: digraph? why not just application.png ? or?

Comment (elijahbenizzy) [reply examples/validate_examples.py:41]: It's now `statemachine.png`

**Comment by elijahbenizzy** [comment]
More work in this, this I think is good enough for now: https://github.com/DAGWorks-Inc/burr/issues/116

File: examples/blog_post/blog.ipynb
Comment (skrawcz) [inline examples/blog_post/blog.ipynb:1]: you will need to update links in a few places, e.g. the blog, docs, etc.

File: examples/llm-adventure-game/digraph
Comment (skrawcz) [inline examples/llm-adventure-game/digraph:1]: should this be committed?

**Review by skrawcz — APPROVED** [review]
we need to update youtube with links too.

File: examples/blog_post/blog.ipynb
Comment (elijahbenizzy) [reply examples/blog_post/blog.ipynb:1]: Will do post-release

**Pull request merged**